crates/node/src/catchup.rs

```diff
@@ -9,6 +9,28 @@ use crate::{types, Node};
 mod batch;
 
 impl Node {
+    pub fn derive_executor_public_key(
+        &self,
+        context_id: &calimero_primitives::context::ContextId,
+        executor_public_key: &[u8; 32],
+    ) -> eyre::Result<[u8; 32]> {
+        let handle = self.store.handle();
+
+        // Retrieve ContextIdentity
+        let identity_key =
+            calimero_store::key::ContextIdentity::new(*context_id, *executor_public_key);
+        let identity: calimero_primitives::identity::ContextIdentity =
+            handle.get(&identity_key)?.ok_or_else(|| {
+                eyre::eyre!(
+                    "ContextIdentity not found for context {} and public key {:?}",
+                    context_id,
+                    executor_public_key
+                )
+            })?;
+
+        Ok(identity.public_key)
+    }
```

Comment:
I'm not sure I understand this method

Doesn't it just return the `executor_public_key` that we already have in the best case (passed in args)? and fail otherwise? (which will happen during initial catchup?)

Reply:
This evolved over time and became essentially useless, so I've removed it 👍